sdk/eventhub/azure-eventhub/azure/eventhub/_connection_string_parser.py

```diff
@@ -11,16 +12,16 @@ class EventHubConnectionStringProperties(DictMixin):
     Properties of a connection string.
     """
 
-    def __init__(self, **kwargs):
-        self._fully_qualified_namespace = kwargs.pop("fully_qualified_namespace", None)
-        self._endpoint = kwargs.pop("endpoint", None)
-        self._eventhub_name = kwargs.pop("eventhub_name", None)
-        self._shared_access_signature = kwargs.pop("shared_access_signature", None)
-        self._shared_access_key_name = kwargs.pop("shared_access_key_name", None)
-        self._shared_access_key = kwargs.pop("shared_access_key", None)
+    def __init__(self, **kwargs: Any) -> None:
+        self._fully_qualified_namespace: Optional[str] = kwargs.pop("fully_qualified_namespace", None)
```

Comment:
we can type all of the kwargs out. For a conn string, will any of these segments be `None`?

Reply:
update, they can be None